snapshot/pkg/apis/crd/v1/types.go

```diff
@@ -270,6 +279,9 @@ func GetSupportedVolumeFromPVSpec(spec *core_v1.PersistentVolumeSpec) string {
 	if spec.Glusterfs != nil {
 		return "glusterfs"
 	}
+	if spec.ISCSI != nil {
```

Comment:
This can conflict with non openebs based iSCSI. Is there any other field in the spec that we can use - like may be provisioner?

Reply:
There is thought process going on to support out-tree `VolumeSource`, we have to find out the generic way to solve this issue, so its helps other providers too.
There is one field called `FlexVolumeSource` for flex volume provisioning , something similar we need to have for external-provisioner to work.

Reply:
Can you link the issue that is tracking the above?

Reply:
https://github.com/kubernetes-incubator/external-storage/issues/574